sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFiltersBase.scala

```diff
@@ -67,18 +65,12 @@ trait OrcFiltersBase {
       }
     }
 
-    val primitiveFields = getPrimitiveFields(schema.fields)
-    if (caseSensitive) {
-      primitiveFields.toMap
-    } else {
-      // Don't consider ambiguity here, i.e. more than one field are matched in case insensitive
-      // mode, just skip pushdown for these fields, they will trigger Exception when reading,
-      // See: SPARK-25175.
-      val dedupPrimitiveFields = primitiveFields
-        .groupBy(_._1.toLowerCase(Locale.ROOT))
-        .filter(_._2.size == 1)
-        .mapValues(_.head._2)
-      CaseInsensitiveMap(dedupPrimitiveFields)
-    }
+    // Different with Parquet case, for case insensitive analysis, we will set
+    // `OrcConf.IS_SCHEMA_EVOLUTION_CASE_SENSITIVE`. So we don't need to worry about
```

Comment:
Out of curiosity, do we have one test to verify the case sensitivity vs insensitivity? Other changes look good

Reply:
We don't have case sensitivity test for ORC filter pushdown. Because the case-insensitive column resolution is done inside ORC library and this is not for a bug, I'm not able to add a regression test.

Reply:
Hm, but we could have one e2e test instead though. E.g. in `OrcQueryTest` by leveraging `stripSparkFilter`, having one table with column `A` and `a` and filters against `a`.

Reply:
Oh, that looks like Parquet ones. Ok, I think I can write a few test. Let me do it tomorrow. Thanks.

Reply:
Yep. +1 for using `stripSparkFilter`.

Reply:
The test was added in #29427. Seems ORC library doesn't process pushed predicate in case-insensitive approach as expected.

That's said, even `OrcConf.IS_SCHEMA_EVOLUTION_CASE_SENSITIVE ` is set to false at Spark ORC datasource, the column names in pushed predicates are matched in case-sensitive way.

This is shown in the added test.